src/System.Net.Http.WinHttpHandler/tests/UnitTests/WinHttpHandlerTest.cs

```diff
@@ -846,6 +846,20 @@ public void SendAsync_NonSecureRequestToServer_ExpectNoClientCertContext()
                 async () => { await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cts.Token); });
         }
 
+        [Fact]
+        public async Task SendAsync_WinHttpOpenReturnsError_ExpectHttpRequestException()
+        {
+            var handler = new WinHttpHandler();
+            var client = new HttpClient(handler);
+            var request = new HttpRequestMessage(HttpMethod.Get, FakeServerEndpoint);
+
+            TestControl.Fail.WinHttpOpen = true;
+
+            Exception ex = await Assert.ThrowsAsync<HttpRequestException>(
+                async () => { await client.SendAsync(request); });
```

Comment:
Yes, the code above you wrote does compile.

But interestingly, if I add surrounding braces to my one line statement, it doesn't compile.  I.e. this does not compile and will generate the error I mentioned:

`Exception ex = await Assert.ThrowsAsync<HttpRequestException>(() => { client.SendAsync(request); } );`